quickwit/quickwit-config/src/source_config/mod.rs

```diff
@@ -229,12 +228,18 @@ pub enum SourceParams {
 }
 
 impl SourceParams {
-    pub fn file<P: AsRef<Path>>(filepath: P) -> Self {
-        Self::File(FileSourceParams::file(filepath))
+    pub fn file_from_uri(uri: Uri) -> Self {
+        Self::File(FileSourceParams {
+            filepath: Some(uri),
+        })
+    }
+
+    pub fn file_from_str<P: AsRef<str>>(filepath: P) -> anyhow::Result<Self> {
```

Comment:
I don't think this is necessary, it's already super easy to go from `&str` to `Uri` and then to a file params object.